ext/datadog_profiling_native_extension/stack_recorder.c

```diff
@@ -1057,3 +1040,17 @@ static VALUE build_profile_stats(profile_slot *slot, long serialization_time_ns,
   for (long unsigned int i = 0; i < VALUE_COUNT(arguments); i += 2) rb_hash_aset(stats_as_hash, arguments[i], arguments[i+1]);
   return stats_as_hash;
 }
+
+static u_int32_t intern_or_raise(struct stack_recorder_state *state, const char *str) {
+  if (str == NULL) {
+    return 0;
+  }
+  ddog_CharSlice char_slice;
+  char_slice.len = strlen(str);
+  char_slice.ptr = str;
```

Comment:
Minor: It may be worth introducing a macro version of this to allow the compiler to hardcode the strlen at compile-time for static strings, like `DDOG_CHARSLICE_C` does.